src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/provideappointeedetails/ProvideAppointeeDetailsAboutToSubmitHandler.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.provideappointeedetails;
+
+import static java.util.Objects.requireNonNull;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.InterlocReviewState;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+@Service
+@Slf4j
+public class ProvideAppointeeDetailsAboutToSubmitHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.ABOUT_TO_SUBMIT)
+                && callback.getEvent() == EventType.PROVIDE_APPOINTEE_DETAILS;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+
+        log.info("Setting interloc review state after provide appointee details event for case id: {}", callback.getCaseDetails().getId());
+
```

Comment:
should there be an error returned if there is already an appointee?

Reply:
That wasn't an AC on the ticket @ryandac so I don't think so. If there is already an appointee then it will just repopulate their details in this event. I'll run past Rosie to see if this is acceptable.